gapis/trace/android/adreno/profiling_data.go

```diff
@@ -294,3 +305,35 @@ func processCounters(ctx context.Context, processor *perfetto.Processor, desc *d
 	}
 	return counters, nil
 }
+
+// For a renderPass leafy group, find its parent group and return it.
+// If the parent groups are not created yet, create them and store in the map.
+func findParentGroup(ctx context.Context, subOrder, cb uint64, groupsMap map[api.CmdSubmissionKey]*service.ProfilingData_GpuSlices_Group, links map[api.CmdSubmissionKey][]api.SubCmdIdx, capture *path.Capture) *service.ProfilingData_GpuSlices_Group {
```

Comment:
This logic seems to be duplicated between adreno and mali, would there be a way to asbtract it in a single place to avoid duplication?

Reply:
Thanks for pointing it out, just done it in a new commit!